### ISSUE TITLE
[Bug]: serial.getPorts return empty even after setDevicePermissionHandler grants the permissions

### ISSUE THREAD

**Issue by belerweb** [electron/electron#32505]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

16.0.0;15.3.5

### What operating system are you using?

macOS

### Operating System Version

12.1

### What arch are you using?

x64

### Expected Behavior

According v16.0.0 release notes: 
Changed Web Serial to use ses.setDevicePermissionHandler to allow developers more control over Web Serial permissioning. #31324 (Also in 15)

I expect ```navigator.serial.getPorts().then(ports=>console.log(ports))``` return all serial port without call ```navigator.serial.requestPort()``` after setDevicePermissionHandler grant the permissions.
 
```
    mainWindow.webContents.session.setPermissionCheckHandler((webContents, permission, requestingOrigin, details) => {
        return true;
    });
    mainWindow.webContents.session.setDevicePermissionHandler((details) => {
        return true;
    });
```

I expect ```navigator.serial.getPorts()``` worked like ```navigator.hid.getDevices()``` worked without call ```navigator.hid.requestDevice()```.

### Actual Behavior

```navigator.serial.getPorts().then(ports=>console.log(ports))``` return empty without call ```navigator.serial.requestPort()``` 

### Testcase Gist URL
https://gist.github.com/d9de1c981ef830835c0e05931dc5f8f4

_No response_

**Comment by codebytere** [comment]
cc @jkleinsc

**Comment by jkleinsc** [comment]
@belerweb what kind of serial devices do you have attached to your machine when testing?  When I attach a legitimate serial device to my machine and run your test it works.

**Comment by belerweb** [comment]
It's vehicle exhaust gas detection equipment. I can't identify its manufacturer.
I don't think it has anything to do with the device.

My gist navigator.serial.requestPort() return 4 devices on my computer, but navigator.serial.getPorts() return empty if i do not call navigator.serial.requestPort() and callback with one device.

And I found that "callback" set by setPermissionCheckHandler and  setDevicePermissionHandler not execute when call navigator.serial.getPorts(). But call navigator.hid.requestDevice(), it executed.

**Comment by jkleinsc** [comment]
I believe what is happening here is due to https://github.com/electron/electron/blob/main/shell/browser/serial/serial_chooser_context.cc#L163, which was copied from this upstream change: https://chromium-review.googlesource.com/c/chromium/src/+/2413176.  Given that we are giving control to the developer as to which ports are being exposed, we can probably skip this check, but there is no guarantee that those ports that fall into this scenario will actually work.

**Comment by MaxMaeder** [comment]
Is this fix in the latest release (18.0.3)? I'm still running into this issue.

**Comment by freewong** [comment]
same issue here. my computer have a COM5 port,but this array ports is empty.
`
document.getElementById('btn_test').addEventListener('click',()=>{

console.log('serial' in navigator);

    navigator.serial.getPorts().then((ports) => {
        console.log(ports);
      });

},false);

`
